eof/scihubclient.py

```diff
@@ -1,76 +1,41 @@
 """sentinelsat based client to get orbit files form scihub.copernicu.eu."""
 
-
-import re
 import logging
 import datetime
 import operator
 import collections
-from typing import NamedTuple, Sequence
+from typing import Sequence
 
-from .products import Sentinel as S1Product
+from .products import SentinelOrbit, Sentinel as S1Product
 
 from sentinelsat import SentinelAPI
 
 
 _log = logging.getLogger(__name__)
 
 
-DATE_FMT = '%Y%m%dT%H%M%S'
-
-
 class ValidityError(ValueError):
     pass
 
 
-class ValidityInfo(NamedTuple):
-    product_id: str
-    generation_date: datetime.datetime
-    start_validity: datetime.datetime
-    stop_validity: datetime.datetime
-
-
-def get_validity_info(products: Sequence[str],
-                      pattern=None) -> Sequence[ValidityInfo]:
-    if pattern is None:
-        # use a generic pattern
-        pattern = re.compile(
-            r'S1\w+_(?P<generation_date>\d{8}T\d{6})_'
-            r'V(?P<start_validity>\d{8}T\d{6})_'
-            r'(?P<stop_validity>\d{8}T\d{6})\w*')
-
-    keys = ('generation_date', 'start_validity', 'stop_validity')
-    out = []
-    for product_id in products:
-        mobj = pattern.match(product_id)
-        if mobj:
-            validity_data = {
-                name: datetime.datetime.strptime(mobj.group(name), DATE_FMT)
-                for name in keys
-            }
-            out.append(ValidityInfo(product_id, **validity_data))
-        else:
-            raise ValueError(
-                f'"{product_id}" does not math the regular expression '
-                f'for validity')
-
-    return out
+def get_validity_info(products: Sequence[str]) -> Sequence[SentinelOrbit]:
```

Comment:
I only realized after a second look that the `SentinelOrbit` class hadd the same functionality as the validity NamedTuple, with just slightly different names (start_time instead of start_validity)